docs_src/getting-started/tools/Ch-CommandLineInterface.md

```diff
@@ -0,0 +1,104 @@
+
+It replaces the need to manually construct complex CURL commands. In particular, it helps with Post/Update requests (example: create a new event, or update a device) because the request body of these requests often include a lot of parameters. EdgeX-CLI provides an "interactive mode" that greatly simplifies the request constructing process. Commands that represent Post/Update requests leverage the interactive mode that opens the user's default editor with preloaded request scaffolding ready to be customized. The interactive mode also works with in-line parameters - allowing in-line values of the CLI command to get loaded in to the request scaffolding. 
+
+EdgeX-CLI is an easy to use helper tool that benefits both EdgeX developers and users. This is the first release of the CLI with future enhancements planned.  Currently, the CLI only works in insecure mode (that is without the EdgeX API gateway protecting EdgeX services), but there is a lot more to come in future EdgeX releases.
+
+## How to get the EdgeX CLI
+
+Download the appropriate compressed file for your operating system containing the edgex-cli executable from [GitHub](https://github.com/edgexfoundry/edgex-cli/releases).
+
+The Edgex-CLI is also publish in snap store (Snapcraft.io). To install the EdgeX CLI snap run:
+
+```
+sudo snap install edgex-cli
+```
+
+You can also get the source-code and then build and install edgex-cli from the source.  Again, find the source code on [GitHub](https://github.com/edgexfoundry/edgex-cli).
+
+**For more information read see the tool's [README](https://github.com/edgexfoundry/edgex-cli/blob/master/README.md).**
+
+## What can the CLI do?
+
+The Edgex CLI covers most of the basic EdgeX Foundry APIs. Using the CLI, users can query all the available devices, profiles, services, events, readings etc. or query for a subset of them based on available flags – for example get devices with a specific name or get all events that belong to specific device.
+
+It also provides the means to create EdgeX objects such as devices, profiles, etc.  Creating can be accomplished in three different ways:  
+- By providing a file that contains the normal create object API request body
+- By using inline parameters
+- Or using interactive mode.
+
+Some of the request bodies could contain a lot parameters and writing all them in one single line may not be convenient.  For these cases, the interactive mode used/or not altogether with inline parameters could be more convenient.
+
+
+Updating EdgeX objects via the CLI can done in two ways:
+ - By providing a file that contains the update API request body that would normally be used in the API call
+ - Or by providing just the name of the object and, if the object with such name exists, enter into editor mode with the preloaded request body where you can change/edit just the fields that need to be modified. 
```

Comment:
s/Or by providing just the name of the object and, if the object with such name exists, enter into editor mode with the preloaded request body where you can change/edit just the fields that need to be modified. /Or by providing just the name of the object. If the object with the specified name exists, the CLI opens an editor with a preloaded request body from the retrieved values, allowing you to change/edit desired fields (except the name & ID).